WalletWasabi/WabiSabi/Models/TransactionSignaturesRequest.cs

```diff
@@ -5,31 +5,5 @@
 
 namespace WalletWasabi.WabiSabi.Models
 {
-	public record TransactionSignaturesRequest(uint256 RoundId, IEnumerable<InputWitnessPair> InputWitnessPairs)
-	{
-		public override int GetHashCode()
-		{
-			int hash = 0;
-
-			foreach (InputWitnessPair item in InputWitnessPairs)
-			{
-				hash = (hash << 4) + item.GetHashCode();
-			}
-
-			return HashCode.Combine(RoundId, hash);
-		}
-
-		public virtual bool Equals(TransactionSignaturesRequest? other)
-		{
-			if (other is null)
-			{
-				return false;
-			}
-
-			bool isEqual = RoundId == other.RoundId
-				&& InputWitnessPairs.SequenceEqual(other.InputWitnessPairs);
-
-			return isEqual;
-		}
-	}
+	public record TransactionSignaturesRequest(uint256 RoundId, uint InputIndex, WitScript Witness);
```

Comment:
@lontivero i didn't want to change this, i think i made it a uint so that it couldn't represent negative values, before learning that they are second class citizens in C#. should this just be an int to avoid the cast elsewhere?

Reply:
No, it is okay as an `unit` because it is used as an index and can never be negative. I don't know where we have to cast it but if that is in our code then it is wrong in those other parts and not here.

Reply:
C# indices are generally `int`, not `uint`, so this is cast before being passed to `AddWitness`. There's no problem with changing that method signature to take a uint, but eventually it will have to 

Previously I was told that using ulongs and uints to exclude negative values is something that is not often done in C# (and i guess it makes sense to prefer ints for the general type of indexes, because subtracting indexes is a common operation, and checking that the result didn't overflow is more subtle than checking that the result isn't negative...)